templates/FFV_form_factors.cpp.in

```diff
@@ -602,6 +604,239 @@ double ScalarC0(double M22, double M32, double m1, double m2, double m3)
 }
 
 
+std::complex<double> FFV_FFS_A1L(double mi, double mj, double mF, double mS, auto SFin, auto SFout)
```

Comment:
Sorry to be pedantic, but do you have some references for these longish expressions? Or did you derive them yourself? If there is some reference, would you mind adding a comment on top of the function?

Reply:
No, these are self-derived.  I derived them in the same way that I derived the original one loop functions, just dropping the assumptions about the masses mi & mj being small, and assuming special cases such as mi==mj, mS or mF==0.

   I can put in a large explanation giving the derivation when I have finished the other general contributions.

Reply:
Thanks, that would be good!

Do you also have Mathematica expressions for these? If so, could you put them into a file in the `test/` subdirectory? Maybe `test/AMuonLoopFunctions.m`? That would be best for future testing and/or comparison and/or re-derivation, if needed. :)

Reply:
We can do test functions with Mathematica?  Then, my Mathematica expressions for the general loop functions can certainly go in there.  Thanks!

Reply:
My long-term intention would be to use your Mathematica expressions to generate data-sets for testing the numerical C++ implementation of these functions.

Of course, we can also test the Mathematica functions themselves, if you like. But having the expressions in some algebraic form would be a perfect first step. :)